docs/examples/01_current_injection.py

```diff
@@ -1,502 +1,694 @@
-#     In nemos, we default to the exponential, other choices such as soft-plus, are allowed. These
-#     choices guarantee convexity, i.e. a single optimal solution.
-#     In principle, one could choose a more complex non-linearity, but convexity is not
-#     guaranteed in general.
+#
+# Now that we've initialized our model with the optimization parameters, we can
+# fit our data! In the previous section, we prepared our model matrix
+# (`predictor`) and target data (`count`), so to fit the model we just need to
+# pass them to the model:
+
+model.fit(predictor, count)
 
 # %%
-# ### 3. The Poisson log-likelihood
-# The last component of the model is the poisson log-likelihood that quantifies how likely it is
-# to observe certain counts for a given firing rate time series.
-# The if $y(t)$ are the spike counts, the equation for the log-likelihood is
-# $$ \sum\_t \log P(y(t) | \lambda(t)) = \sum\_t  y(t) \log(\lambda(t)) - \lambda(t) - \log (y(t)!)\tag{2}$$
-# In nemos, the log-likelihood can be computed by calling the score method passing the predictors and the counts.
-# The method first compute the rate $\lambda(t)$ using (1) and then the likelihood using (2).
+#
+# Now that we've fit our data, we can retrieve the resulting parameters, the
+# analogues of the `weights` and `intercepts` variables used earlier in this
+# notebook. Similar to scikit-learn, these are stored as the `coef_` and
+# `intercept_` attributes:
 
-log_likelihood_0 = model.score(input_feature, counts, score_type="log-likelihood")
-print(f"log-likelihood hard-coded weights: {log_likelihood_0}")
+print(f"firing_rate(t) = exp({model.coef_} * current(t) + {model.intercept_})")
 
 # %%
-# We can learn the  maximum-likelihood (ML) weights by calling the fit method. If we compare the likelihood after
-# fitting, it increased.
-model.fit(input_feature, counts)
-log_likelihood_ML = model.score(input_feature, counts, score_type="log-likelihood")
-print(f"log-likelihood ML weights: {log_likelihood_ML}")
+#
+# Note that `model.coef_` and `model.intercept_` are not scalars, but
+# multi-dimensional:
 
-# %%
-# !!! warning
-#     explain why LBFGS
+print(f"coef_ shape: {model.coef_.shape}")
+print(f"intercept_ shape: {model.intercept_.shape}")
 
 # %%
-# Now that we've fit our data, we can use the model to predict the firing rates and
+#
+# `model.coef_` has shape `(n_neurons, n_features)`, while `model.intercept_`
+# has shape `(n_neurons)`, with those values coming from the model matrix we
+# passed during fitting.
+#
+# It's nice to get the parameters above, but we can't tell how well our model
+# is doing by looking at them. So how should we evaluate our model?
+#
+# First, we can use the model to predict the firing rates and compare that to
+# our observed firing rate, to see which of the phenomena we described in our
+# pynapple analysis our model is able to capture. By calling `predict()` we can
```

Comment:
I suggest that we change "observed firing rate" -> "smoothed spike train" (or something like this) everywhere.

Also I think this sentence can be shortened a bit

```suggestion
# First, we can use the model to predict the firing rates and compare that to
# the smoothed spike train. By calling `predict()` we can
```